searchcore/src/vespa/searchcore/bmcluster/storage_api_rpc_bm_feed_handler.cpp

```diff
@@ -83,10 +83,20 @@ StorageApiRpcBmFeedHandler::~StorageApiRpcBmFeedHandler() = default;
 void
 StorageApiRpcBmFeedHandler::send_rpc(std::shared_ptr<storage::api::StorageCommand> cmd, PendingTracker& pending_tracker)
 {
-    cmd->setSourceIndex(0);
-    cmd->setAddress(*_storage_address);
-    _message_dispatcher->retain(cmd->getMsgId(), pending_tracker);
-    _rpc_client->send_rpc_v1_request(std::move(cmd));
+    auto bucket = cmd->getBucket();
+    if (_distributor) {
```

Comment:
Consider if the handler classes should inherit a common abstract class to share some member variables and code that is currently duplicated around. The same pattern is used in _StorageApiChainBmFeedHandler::send_msg_ and _StorageApiMessageBusBmFeedHandler::send_msg_.

Reply:
Common code has been moved to _StorageApiBmFeedHandlerBase_.